src/resmom/requests.c

```diff
@@ -793,6 +793,8 @@ message_job(job *pjob, enum job_file jft, char *text)
 				usleep(usecs);
 			else {
 				(void)close(fds);
+				if (pstr)
+					free(pstr);
```

Comment:
You have properly initialized pstr to NULL when you declared it, so no need to check for NULL. Just call free().

Reply:
done.